distributed/protocol/serialize.py

```diff
@@ -195,7 +195,9 @@ def check_dask_serializable(x):
     return False
 
 
-def serialize(x, serializers=None, on_error="message", context=None):
+def serialize(
+    x, serializers=None, on_error="message", context=None, iterate_collection=None
```

Comment:
Perhaps, if `iterate_collection=None` isn't supposed to have a distinct meaning from `iterate_collection=False` (which seems to be the case), the default value could be set to `False`?

Reply:
Right - I guess the current logic will use `False` as the default anyway, so we might as well be explicit.  Thanks, good suggestion.

Reply:
Yes, my thinking has been that, this way the intended value space would be documented more clearly.

Reply:
After looking back at the code a bit, I think the `iterate_collection=None` default actually makes sense.  The problem is that we do not need the `iterate_collection = iterate_collection or False` line below.

The behavior we want:

- `iterate_collection=True`: Serialization **always** dumps the elements of the collection **separately**
-  `iterate_collection=False`: Serialization **always** dumps the entire collection **together**
-  `iterate_collection=None` (**default**): The ideal approach is inferred by the existence/location of "pickle" among the list of available serializers.

Reply:
Indeed, that `iterate_collection = iterate_collection or False` was redundant. Thanks for updating the docstring!
One potential style point about `in (list, set, tuple, dict):`. Not sure if there are any firm guidelines about this (`distributed`-wise or Dask organization-wide), but perhaps that should be considered as well (i.e., `in <tuple>` vs `in <set>`).